ga4gh/datamodel/genotype_phenotype.py

```diff
@@ -0,0 +1,467 @@
+            diseases_clause = "(" + " || ".join(diseases) + ")"
+            filters.append(diseases_clause)
+
+        filter = "FILTER (" + " && ".join(filters) + ")"
+        query = query.replace("%FILTER%", filter)
+        query = query.replace("%LOCATION%", location_clause)
+        return query
+
+    def query(self, filter=''):
+        """
+        This is the 'detail' query
+
+        Return a list of dictionaries.
+        Each dict is an [annotation](http://www.w3.org/ns/oa#Annotation)
+        All keys in the dict are predicates of the annotation.
+        All cells in the dict are predicate->object converted to strings.
+
+        If an annotation has a <http://purl.org/oban/association_has_subject>
+        predicate that class is appended to the annotation dict in the
+        'location' property
+        """
+        #
+        annotation_query = """
+        PREFIX OBAN: <http://purl.org/oban/>
+        PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+        SELECT distinct *
+          WHERE {
+          ?s    a OBAN:association .
+          ?s ?p ?o .
+          OPTIONAL {?o  rdfs:label ?label .} .
+        %FILTER%
+        }
+        """
+
+        annotation_query = annotation_query.replace("%FILTER%", filter)
+        # print(annotation_query)
+        results = G2PDataset.g.query(annotation_query)
+
+        if (G2PDataset.record_test):
+            self.test_data(results)
+        rows = [row.asdict() for row in results]
+        for row in rows:
+            for k in row:
+                row[k] = row[k].toPython()
+
+        self.record_more(rows)
+
+        location_query = """
+        PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+             PREFIX OBO: <http://purl.obolibrary.org/obo/>
+        SELECT distinct *
+        WHERE {
+        ?s    a  OBO:SO_0001059  .
+        ?s ?p ?o .
+        OPTIONAL {?o  rdfs:label ?label .} .
+        %FILTER%
+        }
+        """
+
+        location_rows = []
+        for row in rows:
+            if (row['p'] == 'http://purl.org/oban/association_has_subject'):
+                location = row['o']
+                location_query = location_query.replace("%FILTER%",
+                                                        "FILTER (?s = <" +
+                                                        location + ">)")
+                # print(location_query)
+                results = G2PDataset.g.query(location_query)
+                if (G2PDataset.record_test):
+                    self.test_data(results)
+                location_rows = [row.asdict() for row in results]
+                for row in location_rows:
+                    for k in row:
+                        row[k] = row[k].toPython()
+        self.record_more(rows)
+
+        annotation = self.flatten(rows)
+        location = self.flatten(location_rows)
+        annotation['location'] = location
+        return annotation
+
+    def flatten(self, dict):
```

Comment:
Why do we need to do this? Is there any way we can move directly from the results that we get to the GA4GH objects without all the intermediate dicts? I'm finding it hard to follow what's going on here.